### DISCUSSION TITLE
Daisy seed build

### DISCUSSION THREAD

**Discussion by jpnielsen** [eh2k/squares-and-circles#84] [Q&A]
Hello,

If it's not too much work, I would like to run Squares and Circles on a [Daisy Seed rev 4](
https://web.archive.org/web/20240509184715/https://daisy.nyc3.cdn.digitaloceanspaces.com/products/seed/Daisy_Seed_datasheet_v1.1.pdf) (with the AK4556 codec). Is that possible now, with the sourcecode you have made available here ?

The only PCB i have for it right now, is the [MiniVerb](https://benjiaomodular.com/post/2023-02-10-miniverb-v1/) module- so I would have to build something - maybe later  ... including the extras on the new T4.1 O_C board, when that design is available?

Are there any benefits from running O_C on the daisy seed instead of Teensy 4.0/4.1 - longer reverbs due to 64MB RAM, better sound quality from the dedicated audio codec  ... or is it not worth it?

**Comment by eh2k** [comment]
Hi,

here is only the source code of the engines (apps directory) and the third-party dependencies I have used (libs directory).
Some of the third-party libraries have been modified.

It is relatively easy to get individual engines running on other targets. Especially for the Miniverb you could take a close look at the apps/FX directory.

However, the complete S&C firmware consists of two parts. The S&C loader which is basically the operating system is ‘closed-source’. In contrast to classic bootloaders, it can run several app instances simultaneously and provides a system API and general functionality like ‘virtual routing, modulations, midi & clock handling and patch management etc’. There are different versions of the ‘squares-and-circles-loader’ (optimized for the specific hardware, like Teensy 4.0/4.1 or daisy patch submodule). The engines/apps can be developed device-independent for all supported targets and can be executed as binaries directly on them (e.g. from an SD card or from a defined area in the flash memory).

Regarding the T4.1 O_C. I have tested the Rev2 prototype - and more or less ported the S&C loader to it. The Rev2 board had issues with the external ADC - as far as I can tell, Paul wanted to fix this with Rev3. I also made a few requests/suggestions for changes - e.g. optional resistors so that the ADC or DAC can be modified to the -5..5V range. The Rev2 8xDAC was designed for -3..7v and was mostly intended for cv output. Apart from that, I have never seen the Rev3 and don't know when or if the T4.1 O_C will be available.

To be honest, I have put my plan to complete a Rev2 S&C module based on the daisy patch submodule on hold because of the T4.1 O_C. The motherboard with the T4.1 would actually be a good alternative to the daisy patch submodule (mainly because of the 8xDAC, the SD card directly on the Teensy 4.1 and some other details). But DIY in this case will probably be advanced-level.
Regarding the UI, I have a slightly different idea (which would make the most sense to me, having used S&C on O_C for more than 2 years).

If you want to do something with samples, RAM and SD card are useful. I had seen reverbs that were programmed with faust (or pure-data) - which required quite a lot of RAM. (Teensy 4.0 without extra RAM is not optimal for that use-cases).
